### ISSUE TITLE
Infinite loop when a dependency is missing in proto file

### ISSUE THREAD

**Issue by whiver** [os72/protobuf-dynamic#8]
I have the following method:

```
public static Descriptors.Descriptor parseProto (String protoPath, String messageType) throws IOException, Descriptors.DescriptorValidationException {
        FileInputStream schemaFile = new FileInputStream(protoPath);
        DynamicSchema schema = DynamicSchema.parseFrom(schemaFile);
        return schema.getMessageDescriptor(messageType);
    }
```

But when executed, the program is caught in an infinite loop in DynamicSchema.init() on line 219:
```
while (fileDescMap.size() < fileDescSet.getFileCount()) {
```

afaik, the condition is always true since `fileDescMap` is cleared just before and nothing is ever added into it, and `fileDescSet` has elements (1 in my case) which are never removed.

EDIT: I just tried with your sample schema (Schema2.desc), it is working well, but I guess it is because there is no dependency. I am personally using the example proto from Google: https://github.com/google/protobuf/blob/master/examples/addressbook.proto

EDIT 2: I finally found the problem: it came from my .proto file. If an import statement is not available (external statement), it will keep looping trying to add it to the set. It is a problem in my case as I am not aware of the proto file that will be parsed, I don't want the parser to freeze my app. Maybe throwing an exception in case of missing dependency?

**Comment by os72** [comment]
Ack. Sorry I missed this, github stopped showing issue updates on my main page (I'd like to know how to re-enable that..)

Can you share a schema to reproduce this?

**Comment by os72** [comment]
Both `Schema1.desc` and `Schema2.desc` (used in tests) have imports and the resolution works fine

What are your steps to reproduce? Is it possible you compiled without `--include_imports`? Maybe we could add a check for missing imports

**Comment by os72** [comment]
Released 0.9.4

**Comment by whiver** [comment]
Hello, sorry for the delay too!
Yes indeed, I compiled without `--include_imports`, and it was producing an infinite loop in that case. But I see you have fixed it, thanks ! :)
